File: scripts/src/bin/write_palette_table.rs
use flamerial_scripts::*;

fn main() {
    println!("<table align=\"center\">");
    println!("    <thead>");
    println!("        <tr>");
    for heading in ["Preview", "Name", "HEX", "RGB"] {
        println!("            <th>{heading}</th>");
    }
    println!("        </tr>");
    println!("    </thead>");
    println!("    <tbody>");
    for color in FLAMERIAL_PALETTE.all_colors() {
        println!("        <tr>");
        println!(
            "            <td align=\"center\"><img alt=\"\" src=\"assets/colors/{}.png\" /></td>",
            color.snake_case_name()
        );
        println!("            <td>{}</td>", color.title_case_name());
        println!("            <td><code>{}</code></td>", color.hex_string());
        println!("            <td><code>{}</code></td>", color.rgb_string());
        println!("        </tr>");
    }
    println!("    </tbody>");
    println!("</table>");
    println!(
        "<p align=\"center\"><strong>Caption:</strong> the colors that compose the theme palette.</p>"
    );
}

File: scripts/write_palette_table.rb
require_relative 'lib.rb'

puts '<table align="center">'
puts '  <thead>'
puts '    <tr>'
puts '      <th>Preview</th>'
puts '      <th>Name</th>'
puts '      <th>HEX</th>'
puts '      <th>RGB</th>'
puts '    </tr>'
puts '  </thead>'
puts '  <tbody>'
for key, color in $colors
  puts '    <tr>'
  puts "      <td><img alt=\"\" src=\"assets/#{key}.png\" /></td>"
  puts "      <td>#{key.to_s.split.map(&:capitalize).join ' '}</td>"
  puts "      <td><code>#{color.hex_string}</code></td>"
  puts "      <td><code>#{color.rgb_string}</code></td>"
  puts '    </tr>'
end
puts '  </tbody>'
puts '</table>'
puts '<p align="center"><strong>Caption:</strong> the colors that compose the theme palette.</p>'
